From 867e4e26bd8a96f9bab8ef581ba7c1382e89dcca Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:28:04 -0400
Subject: [tools] perf/evsel: Introduce perf_evsel__prev() method

Message-id: <1408462094-14194-316-git-send-email-jolsa@redhat.com>
Patchwork-id: 88270
O-Subject: [PATCH RHEL7.1 BZ1131394 315/325] perf evsel: Introduce perf_evsel__prev() method
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit d87fcb4a2d990ba2de9284ede84a816c5066d54b
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed Nov 13 15:56:40 2013 -0300

description
===========
Just one use so far, on the hists browser, for completeness since there
we use perf_evlist__{first,last} and perf_evsel__next() for handling the
TAB and UNTAB keys.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/ui/browsers/hists.c b/tools/perf/ui/browsers/hists.c
index 089fd37..a440e03 100644
--- a/tools/perf/ui/browsers/hists.c
+++ b/tools/perf/ui/browsers/hists.c
@@ -1855,7 +1855,7 @@ browse_hists:
     if (pos->node.prev == &evlist->entries)
      pos = perf_evlist__last(evlist);
     else
-     pos = list_entry(pos->node.prev, struct perf_evsel, node);
+     pos = perf_evsel__prev(pos);
     goto browse_hists;
    case K_ESC:
     if (!ui_browser__dialog_yesno(&menu->b,
diff --git a/tools/perf/util/evsel.h b/tools/perf/util/evsel.h
index f502965..1ea7c92 100644
--- a/tools/perf/util/evsel.h
+++ b/tools/perf/util/evsel.h
@@ -279,6 +279,11 @@ static inline struct perf_evsel *perf_evsel__next(struct perf_evsel *evsel)
  return list_entry(evsel->node.next, struct perf_evsel, node);
 }
 
+static inline struct perf_evsel *perf_evsel__prev(struct perf_evsel *evsel)
+{
+ return list_entry(evsel->node.prev, struct perf_evsel, node);
+}
+
 /**
  * perf_evsel__is_group_leader - Return whether given evsel is a leader event
  *
-- 
1.7.1